### DISCUSSION TITLE
ansys-fluent-core install error - h5py/HDF5

### DISCUSSION THREAD

**Discussion by amscosta** [ansys/pyfluent#1041] [Q&A] [answered]
Hello 
I am having the following issue when installing ansys-fluent-core :
(pyenv) C:\Users\User>python -m pip install ansys-fluent-core
Collecting ansys-fluent-core
  Using cached ansys_fluent_core-0.11.0-py3-none-any.whl (3.5 MB)
Collecting ansys-api-fluent~=0.3
  Using cached ansys_api_fluent-0.3.4-py3-none-any.whl (98 kB)
Collecting ansys-platform-instancemanagement~=1.0
  Using cached ansys_platform_instancemanagement-1.0.2-py3-none-any.whl (16 kB)
Collecting grpcio>=1.30.0
  Using cached grpcio-1.50.0-cp311-cp311-win_amd64.whl (3.6 MB)
Collecting numpy>=1.21.5
  Using cached numpy-1.23.4-cp311-cp311-win_amd64.whl (14.6 MB)
Collecting protobuf==3.20.1
  Using cached protobuf-3.20.1-py2.py3-none-any.whl (162 kB)
Collecting appdirs>=1.4.0
  Using cached appdirs-1.4.4-py2.py3-none-any.whl (9.6 kB)
Collecting pandas>=1.1.5
  Using cached pandas-1.5.1-cp311-cp311-win_amd64.whl (10.3 MB)
Collecting h5py>=3.7.0
  Using cached h5py-3.7.0.tar.gz (392 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting ansys-api-fluent~=0.3
  Using cached ansys_api_fluent-0.3.3-py3-none-any.whl (97 kB)
  Using cached ansys_api_fluent-0.3.2-py3-none-any.whl (95 kB)
Collecting importlib-metadata>=4.0
  Using cached importlib_metadata-5.0.0-py3-none-any.whl (21 kB)
Collecting ansys-api-platform-instancemanagement>=1.0.0b3
  Using cached ansys_api_platform_instancemanagement-1.0.0b3-py3-none-any.whl (6.6 kB)
Collecting six>=1.5.2
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting python-dateutil>=2.8.1
  Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
Collecting pytz>=2020.1
  Using cached pytz-2022.5-py2.py3-none-any.whl (500 kB)
Collecting googleapis-common-protos
  Using cached googleapis_common_protos-1.56.4-py2.py3-none-any.whl (211 kB)
Collecting protoc-gen-swagger
  Using cached protoc_gen_swagger-0.1.0-py2.py3-none-any.whl (9.4 kB)
Collecting zipp>=0.5
  Using cached zipp-3.10.0-py3-none-any.whl (6.2 kB)
Building wheels for collected packages: h5py
  Building wheel for h5py (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for h5py (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [72 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-311
      creating build\lib.win-amd64-cpython-311\h5py
      copying h5py\h5py_warnings.py -> build\lib.win-amd64-cpython-311\h5py
      copying h5py\ipy_completer.py -> build\lib.win-amd64-cpython-311\h5py
      copying h5py\version.py -> build\lib.win-amd64-cpython-311\h5py
      copying h5py\__init__.py -> build\lib.win-amd64-cpython-311\h5py
      creating build\lib.win-amd64-cpython-311\h5py\_hl
      copying h5py\_hl\attrs.py -> build\lib.win-amd64-cpython-311\h5py\_hl
      copying h5py\_hl\base.py -> build\lib.win-amd64-cpython-311\h5py\_hl
      copying h5py\_hl\compat.py -> build\lib.win-amd64-cpython-311\h5py\_hl
      copying h5py\_hl\dataset.py -> build\lib.win-amd64-cpython-311\h5py\_hl
      copying h5py\_hl\datatype.py -> build\lib.win-amd64-cpython-311\h5py\_hl
      copying h5py\_hl\dims.py -> build\lib.win-amd64-cpython-311\h5py\_hl
      copying h5py\_hl\files.py -> build\lib.win-amd64-cpython-311\h5py\_hl
      copying h5py\_hl\filters.py -> build\lib.win-amd64-cpython-311\h5py\_hl
      copying h5py\_hl\group.py -> build\lib.win-amd64-cpython-311\h5py\_hl
      copying h5py\_hl\selections.py -> build\lib.win-amd64-cpython-311\h5py\_hl
      copying h5py\_hl\selections2.py -> build\lib.win-amd64-cpython-311\h5py\_hl
      copying h5py\_hl\vds.py -> build\lib.win-amd64-cpython-311\h5py\_hl
      copying h5py\_hl\__init__.py -> build\lib.win-amd64-cpython-311\h5py\_hl
      creating build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\common.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\conftest.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_attribute_create.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_attrs.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_attrs_data.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_base.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_big_endian_file.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_completions.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_dataset.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_dataset_getitem.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_dataset_swmr.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_datatype.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_dimension_scales.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_dims_dimensionproxy.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_dtype.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_errors.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_file.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_file2.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_file_alignment.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_file_image.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_filters.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_group.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_h5.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_h5d_direct_chunk.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_h5f.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_h5o.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_h5p.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_h5pl.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_h5t.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_objects.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_selections.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\test_slicing.py -> build\lib.win-amd64-cpython-311\h5py\tests
      copying h5py\tests\__init__.py -> build\lib.win-amd64-cpython-311\h5py\tests
      creating build\lib.win-amd64-cpython-311\h5py\tests\data_files
      copying h5py\tests\data_files\__init__.py -> build\lib.win-amd64-cpython-311\h5py\tests\data_files
      creating build\lib.win-amd64-cpython-311\h5py\tests\test_vds
      copying h5py\tests\test_vds\test_highlevel_vds.py -> build\lib.win-amd64-cpython-311\h5py\tests\test_vds
      copying h5py\tests\test_vds\test_lowlevel_vds.py -> build\lib.win-amd64-cpython-311\h5py\tests\test_vds
      copying h5py\tests\test_vds\test_virtual_source.py -> build\lib.win-amd64-cpython-311\h5py\tests\test_vds
      copying h5py\tests\test_vds\__init__.py -> build\lib.win-amd64-cpython-311\h5py\tests\test_vds
      copying h5py\tests\data_files\vlen_string_dset.h5 -> build\lib.win-amd64-cpython-311\h5py\tests\data_files
      copying h5py\tests\data_files\vlen_string_dset_utc.h5 -> build\lib.win-amd64-cpython-311\h5py\tests\data_files
      copying h5py\tests\data_files\vlen_string_s390x.h5 -> build\lib.win-amd64-cpython-311\h5py\tests\data_files
      running build_ext
      Loading library to get build settings and version: hdf5.dll
      error: Unable to load dependency HDF5, make sure HDF5 is installed properly
      error: Could not find module 'hdf5.dll' (or one of its dependencies). Try using the full path with constructor syntax.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for h5py
Failed to build h5py
ERROR: Could not build wheels for h5py, which is required to install pyproject.toml-based projects

(pyenv) C:\Users\User>

Any guideline to avoid this error ?

**Comment by dnwillia-work** [comment]
Yeah, it's strange it's trying to build the wheel for h5py (I think). 

It seems to be stopping when installing h5py.  Are you able to pip install that?  It seems to be working ok for me: 

```
D:\ANSYSDev\NoBackup\pyansys\ustest>where python
d:\ANSYSDev\NoBackup\tools\Python310\python.exe

D:\ANSYSDev\NoBackup\pyansys\ustest>python --version
Python 3.10.8

D:\ANSYSDev\NoBackup\pyansys\ustest>python -m venv venvtest

D:\ANSYSDev\NoBackup\pyansys\ustest>venvtest\Scripts\activate

(venvtest) D:\ANSYSDev\NoBackup\pyansys\ustest>python -m pip install h5py
Collecting h5py
  Using cached h5py-3.7.0-cp310-cp310-win_amd64.whl (2.6 MB)
Collecting numpy>=1.14.5
  Using cached numpy-1.23.5-cp310-cp310-win_amd64.whl (14.6 MB)
Installing collected packages: numpy, h5py
Successfully installed h5py-3.7.0 numpy-1.23.5

[notice] A new release of pip available: 22.2.2 -> 22.3.1
[notice] To update, run: python.exe -m pip install --upgrade pip

(venvtest) D:\ANSYSDev\NoBackup\pyansys\ustest>
```
Are you sure you are working with a clean local environment?

**Comment by mkundu1** [comment, marked as answer]
The issue is for Python 3.11 (duplicate of #1147). Our supported versions are Python 3.7 - 3.10 as per README.

**Comment by dnwillia-work** [comment]
Yeah, well spotted.  You can see it here `lib.win-amd64-cpython-311` <<<

**Reply by amscosta** [reply]
Thanks a lot.
